common/client/transaction_sender_test.go

```diff
@@ -17,37 +17,65 @@ import (
 )
 
 type sendTxMultiNode struct {
-	*MultiNode[types.ID, SendTxRPCClient[any]]
+	*MultiNode[types.ID, SendTxRPCClient[any, *sendTxResult]]
 }
 
 type sendTxRPC struct {
 	sendTxRun func(args mock.Arguments)
 	sendTxErr error
 }
 
-var _ SendTxRPCClient[any] = (*sendTxRPC)(nil)
+type sendTxResult struct {
+	err   error
+	txErr error
+	code  SendTxReturnCode
+}
+
+var _ SendTxResult = (*sendTxResult)(nil)
+
+func NewSendTxResult(err error) *sendTxResult {
+	result := &sendTxResult{
+		err:   err,
+		txErr: err,
+	}
+	return result
+}
+
+func (r *sendTxResult) Error() error {
+	return r.err
+}
+
+func (r *sendTxResult) TxError() error {
+	return r.txErr
+}
+
+func (r *sendTxResult) Code() SendTxReturnCode {
+	return r.code
+}
+
+var _ SendTxRPCClient[any, *sendTxResult] = (*sendTxRPC)(nil)
 
 func newSendTxRPC(sendTxErr error, sendTxRun func(args mock.Arguments)) *sendTxRPC {
 	return &sendTxRPC{sendTxErr: sendTxErr, sendTxRun: sendTxRun}
 }
 
-func (rpc *sendTxRPC) SendTransaction(ctx context.Context, _ any) error {
+func (rpc *sendTxRPC) SendTransaction(ctx context.Context, _ any) *sendTxResult {
 	if rpc.sendTxRun != nil {
 		rpc.sendTxRun(mock.Arguments{ctx})
 	}
-	return rpc.sendTxErr
+	return &sendTxResult{err: rpc.sendTxErr, txErr: rpc.sendTxErr, code: classifySendTxError(nil, rpc.sendTxErr)}
 }
 
 func newTestTransactionSender(t *testing.T, chainID types.ID, lggr logger.Logger,
-	nodes []Node[types.ID, SendTxRPCClient[any]],
-	sendOnlyNodes []SendOnlyNode[types.ID, SendTxRPCClient[any]],
-) (*sendTxMultiNode, *TransactionSender[any, types.ID, SendTxRPCClient[any]]) {
-	mn := sendTxMultiNode{NewMultiNode[types.ID, SendTxRPCClient[any]](
+	nodes []Node[types.ID, SendTxRPCClient[any, *sendTxResult]],
```

Comment:
consider creating a type alias for `SendTxRPCClient[any, *sendTxResult]` to make type defintion shorter